### PR TITLE
Feat/basic daemon impl

### PR DESCRIPTION
**Pull request by filberol**

### CODE REVIEW

File: motorsd/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+project(motorsd C)
+
+file(GLOB_RECURSE SOURCES "${CMAKE_CURRENT_LIST_DIR}/src/*.c")
+file(GLOB_RECURSE HEADERS "${CMAKE_CURRENT_LIST_DIR}/include/*.h")
+
+include_directories(include)
+
+set(SOURCES src/daemon.c src/command_parser.c src/motor_invoker.c include/motor_invoker.h)
```
Comment (Moleus) [inline motorsd/CMakeLists.txt:8]: We don't need to use `GLOB_RECURSE SOURCES` if we manually specify source files here

File: motorsd/include/daemon.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#ifndef MOTORLAND_DAEMON_H
```
Comment (Moleus) [inline motorsd/include/daemon.h:1]: In our project we use `#pragma once`

File: motorsd/include/daemon_utils.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#include <stddef.h>
+
+typedef struct {
+    uint8_t motorIndex;
+    uint16_t numOfSteps;
+    uint8_t speed;
+    uint8_t checksum;
+} MotorCommand;
+
+MotorCommand parseMotorCommand(const uint8_t* byteOrder, size_t size);
```
Comment (Moleus) [inline motorsd/include/daemon_utils.h:14]: if this method is public API it should be prefixed with module name, such as `motorsd_parse_motor_command`

Comment (Moleus) [inline motorsd/include/daemon_utils.h:14]: Also, follow snake_case convention

File: motorsd/src/command_parser.c [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#include "daemon_utils.h"
+
+MotorCommand parseMotorCommand(const uint8_t* byteOrder, size_t size) {
```
Comment (Moleus) [inline motorsd/src/command_parser.c:3]: What does `byteOrder` mean?

Comment (Moleus) [reply motorsd/src/command_parser.c:3]: use snake_case for all but Structures

File: motorsd/src/daemon.c [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        size_t bytesRead = recv(clientFd, buffer, sizeof(buffer), 0);
+        if (bytesRead < 0) {
+            perror("Error receiving message from socket");
+            close(clientFd);
+            close(socFd);
+            exit(EXIT_FAILURE);
+        }
+
+        MotorCommand command = parseMotorCommand((const uint8_t *) &bytesRead, sizeof(buffer));
+
```
Comment (Moleus) [inline motorsd/src/daemon.c:68]: In the following updates we will call `process_motor_command(command)` from here, am I right?

File: motorsd/test/test.c [hunk 6]
```diff
@@ -0,0 +1,49 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+#define SOCKET_PATH "/tmp/motorsd_socket"
+
+int main(int argc, char *argv[]) {
```
Comment (Moleus) [inline motorsd/test/test.c:9]: Can u highlight somehow that this test represents a client connection to the server?

**Review by Moleus — CHANGES_REQUESTED** [review]
Please, make the following changes

**Review by Moleus — DISMISSED** [review]

**Comment by Moleus** [comment]
Can we merge this one?

**Review by ThatDraenGuy — APPROVED** [review]
LGTM

**Pull request merged**